tests/tasks/setup/eks/awscli-dp.yaml

```diff
@@ -0,0 +1,87 @@
+    default: "10"
+    description: The maximum number of nodes in the cluster.
+  - name: endpoint
+    default: "https://api.wesley.amazonaws.com"
+  - name: host-cluster-node-role-arn
+    description: arn of the hostcluster node role. This tightly coupled to code here  - https://github.com/awslabs/kubernetes-iteration-toolkit/blob/3ed1bbd47f7b8f111208e977acaa3edfa1834ca8/substrate/pkg/controller/substrate/cluster/addons/karpenter.go#L52 so if it's changed there, it should be changed here. This helps us to avoid creating a separate noderole for nodegroups.
+  workspaces:
+  - name: config
+    description: |
+      A workspace into which a kubeconfig file called `kubeconfig` will be written that will contain the information required to access the cluster. The `kubeconfig` will expect to use [aws-iam-authenticator](https://github.com/kubernetes-sigs/aws-iam-authenticator/) to authenticate, so in order for it to be used it must be run in a container which contains both `kubectl` and `aws-iam-authenticator`.
+  steps:
+  - name: create-nodegroup
+    image: amazon/aws-cli
+    workingDir: $(workspaces.config.path)
+    script: |
+      ssh_key_name=$(params.region)-nodegroups-ssh-key
+      rm -f $HOME/.ssh/$SSH_KEY_NAME.pem
+      if [[ "$(aws ec2 --region $(params.region) describe-key-pairs --key-names $ssh_key_name --query 'KeyPairs[?KeyName==`'$ssh_key_name'`].KeyName' --output text)" -eq 0 ]]; then 
+        echo "KeyPair not found"; 
+        aws ec2 create-key-pair --region $(params.region) --key-name $ssh_key_name --query 'KeyMaterial' --output text > $HOME/.ssh/$SSH_KEY_NAME.pem
+      else 
+        echo "KeyPair already exists"; 
+      fi
+      
+      NG_SUBNETS=$(aws eks --endpoint $(params.endpoint) describe-cluster --name $(params.cluster-name) \
+      --query cluster.resourcesVpcConfig.subnetIds --output text \
+      --region $(params.region))
+
+      export NG_NAME=NG-$(params.cluster-name)-$(params.region)-$(date +%Y%m%d-%H%M%S)
+      
+      #create node group with SSH
+      aws eks --endpoint $(params.endpoint) create-nodegroup \
+      --cluster-name $(params.cluster-name) \
+      --nodegroup-name $NG_NAME \
+      --node-role $(params.host-cluster-node-role-arn) \
+      --region $(params.region) \
+      --remote-access ec2SshKey=$ssh_key_name \
+      --scaling-config minSize=$(params.min-nodes),maxSize=$(params.max-nodes),desiredSize=$(params.desired-nodes) \
+      --subnets $NG_SUBNETS
+
+      while [[ "$(aws eks --endpoint $(params.endpoint) --region $(params.region) describe-nodegroup --cluster-name $(params.cluster-name) --nodegroup-name $NG_NAME --query nodegroup.status --output text)" == "CREATING" ]]
+      do
+      echo "$NG_NAME is "CREATING" at $(date)"
+      sleep 10
+      done
+      aws eks --endpoint $(params.endpoint) update-kubeconfig --name $(params.cluster-name) --region $(params.region)
+      cp  /root/.kube/config $(workspaces.config.path)/kubeconfig
+  - name: validate-nodes
```

Comment:
Do we need to separate this into two steps? It will introduce extra overhead for spinning up new container.

Reply:
The overhead should be very low as it's another pod, unless I am missing something?
This will give us an individual task to validate nodes joining and later we can enhance to check time taken for nodes to join or some additional checks that can pass/fail the test.